### ISSUE TITLE
500 Internal Server Error recurrent issue

### ISSUE THREAD

**Issue by RaedsLab** [verdaccio/verdaccio#1025]
**Describe the bug**
When running `yarn install` on a project that has momentjs as a dependency, I often get this error: 

```
******** react-ui ...
yarn install v1.3.2
[1/4] Resolving packages...
[2/4] Fetching packages...
[91mwarning There appears to be trouble with our server. Retrying...
[0m[91mwarning There appears to be trouble with our server. Retrying...
[0m[91mwarning There appears to be trouble with our server. Retrying...
[0m[91mwarning There appears to be trouble with our server. Retrying...
[0m[91mwarning There appears to be trouble with our server. Retrying...
[0m[91mwarning There appears to be trouble with our server. Retrying...
[0m[91mwarning There appears to be trouble with our server. Retrying...
[0m[91mwarning There appears to be trouble with our server. Retrying...
[0m[91merror An unexpected error occurred: "http://verdaccio.internal/moment/-/moment-2.22.1.tgz: Request failed \"500 Internal Server Error\"".
[0minfo If you think this is a bug, please open a bug report with the information provided in "/opt/source/react-ui/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**To Reproduce**
1. Add moment js as a dependency
2. Run `yarn install` 
3. See error

**Expected behavior**
Install to go through without error.

**Screenshots**

**Docker || Kubernetes (please complete the following information):**
 - Docker verdaccio tag: verdaccio:3.8.1

**Configuration File (cat ~/.config/verdaccio/config.yaml)** 

```yaml
storage: /verdaccio/storage

auth:
  htpasswd:
    file: /verdaccio/conf/htpasswd

uplinks:
  npmjs:
    url: https://registry.npmjs.org/
    cache: false

packages:
  '@*/*':
    access: $all
    publish: $authenticated
    proxy: npmjs

  '**':
    access: $all

    publish: $authenticated

    proxy: npmjs

logs:
  - {type: stdout, format: pretty, level: http}
```

**Comment by juanpicado** [comment]
@RaedsLab server log?

**Comment by RaedsLab** [comment]
Thank you for the response, these are not momentjs related logs, but it seems to happen with other packages as well 


```
http <-- 200, user: null(172.25.0.1), req: 'GET /rc-notification/-/rc-notification-3.1.1.tgz', bytes: 0/89832
http <-- 200, user: null(172.25.0.1), req: 'GET /rc-input-number/-/rc-input-number-4.0.7.tgz', bytes: 0/89730
http <-- 200, user: null(172.25.0.1), req: 'GET /rc-input-number/-/rc-input-number-4.0.7.tgz', bytes: 0/89730
error--- unexpected error: ESOCKETTIMEDOUT
Error: ESOCKETTIMEDOUT
    at ClientRequest.<anonymous> (/usr/local/app/node_modules/request/request.js:812:19)
    at Object.onceWrapper (events.js:273:13)
    at ClientRequest.emit (events.js:182:13)
    at TLSSocket.emitRequestTimeout (_http_client.js:652:40)
    at Object.onceWrapper (events.js:273:13)
    at TLSSocket.emit (events.js:182:13)
    at TLSSocket.Socket._onTimeout (net.js:448:8)
    at ontimeout (timers.js:427:11)
    at tryOnTimeout (timers.js:289:5)
    at listOnTimeout (timers.js:252:5)
http <-- 500, user: null(172.25.0.1), req: 'GET /rc-notification/-/rc-notification-3.1.1.tgz', error: internal server error
warn --> host registry.npmjs.org is now offline
error--- unexpected error: ESOCKETTIMEDOUT
Error: ESOCKETTIMEDOUT
    at ClientRequest.<anonymous> (/usr/local/app/node_modules/request/request.js:812:19)
    at Object.onceWrapper (events.js:273:13)
    at ClientRequest.emit (events.js:182:13)
    at TLSSocket.emitRequestTimeout (_http_client.js:652:40)
    at Object.onceWrapper (events.js:273:13)
    at TLSSocket.emit (events.js:182:13)
    at TLSSocket.Socket._onTimeout (net.js:448:8)
    at ontimeout (timers.js:427:11)
    at tryOnTimeout (timers.js:289:5)
    at listOnTimeout (timers.js:252:5)
http <-- 500, user: null(172.25.0.1), req: 'GET /rc-collapse/-/rc-collapse-1.9.2.tgz', error: internal server error
error--- unexpected error: ESOCKETTIMEDOUT
Error: ESOCKETTIMEDOUT
    at ClientRequest.<anonymous> (/usr/local/app/node_modules/request/request.js:812:19)
    at Object.onceWrapper (events.js:273:13)
    at ClientRequest.emit (events.js:182:13)
    at TLSSocket.emitRequestTimeout (_http_client.js:652:40)
    at Object.onceWrapper (events.js:273:13)
    at TLSSocket.emit (events.js:182:13)
    at TLSSocket.Socket._onTimeout (net.js:448:8)
    at ontimeout (timers.js:427:11)
    at tryOnTimeout (timers.js:289:5)
    at listOnTimeout (timers.js:252:5)
http <-- 500, user: null(172.25.0.1), req: 'GET /rc-editor-mention/-/rc-editor-mention-1.1.7.tgz', error: internal server error
error--- unexpected error: ESOCKETTIMEDOUT
Error: ESOCKETTIMEDOUT
    at ClientRequest.<anonymous> (/usr/local/app/node_modules/request/request.js:812:19)
    at Object.onceWrapper (events.js:273:13)
    at ClientRequest.emit (events.js:182:13)
    at TLSSocket.emitRequestTimeout (_http_client.js:652:40)
    at Object.onceWrapper (events.js:273:13)
    at TLSSocket.emit (events.js:182:13)
    at TLSSocket.Socket._onTimeout (net.js:448:8)
    at ontimeout (timers.js:427:11)
    at tryOnTimeout (timers.js:289:5)
    at listOnTimeout (timers.js:252:5)
http <-- 500, user: null(172.25.0.1), req: 'GET /rc-input-number/-/rc-input-number-4.0.7.tgz', error: internal server error
http <-- 500, user: null(172.25.0.1), req: 'GET /linebreak/-/linebreak-0.3.0.tgz', error: uplink is offline
```

**Comment by juanpicado** [comment]
It seems related to https://github.com/verdaccio/verdaccio/issues/877
Perhaps https://github.com/verdaccio/verdaccio/issues/720

**Comment by RaedsLab** [comment]
I think it could be the same as #877 indeed, restarting the docker usually helps.

**Comment by tfrijsewijk** [comment]
I _think_ I have the same issue.. Although I don't get an error my virtual machine just hangs (at everything, I can't even SSH into it or login through the virtual console).

My CPU usage spikes at 100% with 1 core, at 50% with 2 cores and 25% with 4 cores.

It's a freshly installed Debian 9. I installed screen, curl, sudo and node10. Small packages with `yarn add uuid` and sometimes even `yarn add @angular/cli` work. But generating an angular library with `ng generate library my-lib` always crashes the server hosting verdaccio.

The machine has 2gb memory, 20GB disk space SSD and quad core 2.4Ghz.

I also tried verdaccio 4 alpha 0
